scroll-animations-1/Overview.bs

```diff
@@ -648,35 +653,86 @@ The procedure to calculate <dfn>effective scroll range</dfn> of a
+    from below:
+
+<div class="switch">
+
+:   If <var>current scroll offset</var> is less than [=effective start offset=]:
+::  The [=phase=] is [=before phase|before=]
+
+:   If <var>current scroll offset</var> is greater than or equal to [=effective
+    end offset=] <em>and</em> [=effective end offset=] is less than the maximum
+    scroll offset of {{source}} in {{orientation}}:
+
+::  The [=phase=] is [=after phase|after=]
+
+Note: In web animations, in general ranges are normally exclusive of their end
+point. But there is an exception here for the scroll timeline active range as it
+may in some cases be inclusive of its end. In particular if the timeline end
+offset is the maximum scroll offset we include it in active range because it is
+not possible for user to scroll passed this point and not including this value
+in the active range would leave to animations that would not be active at the
+very last scroll position.
+
+:   Otherwise,
+::  The [=phase=] is [=active phase|active=].
+
+</div>
+
 ### The current time of a {{ScrollTimeline}} ### {#current-time-algorithm}
 
 The [=current time=] of a {{ScrollTimeline}} is calculated as follows:
 
-1.  If {{source}} is null, does not currently have a [=CSS layout box=], or if
-    its layout box is not a [=scroll container=], return an unresolved time
-    value.
+1.  If <em>any</em> of the following are true:
+
+    * {{source}} is null, or
+    * {{source}} does not currently have a [=CSS layout box=], or
+    * {{source}}'s layout box is not a [=scroll container=], or
+    * [=effective scroll range=] is null.
+
+    The [=current time=] is an unresolved time value and abort remaining steps.
 
-1.  Otherwise, let <var>current scroll offset</var> be the current scroll offset
-    of {{source}} in the direction specified by {{orientation}}.
+1.  Let <var>current scroll offset</var> be the current scroll offset of
+    {{source}} in the direction specified by {{orientation}}.
 
-1.  If [=effective scroll range=] is null, return an unresolved time value.
+1.  The [=current time=] is the result corresponding to the first matching
+    condition from below:
 
-1.  If <var>current scroll offset</var> is less than [=effective start offset=],
-    return 0.
+<div class="switch">
 
-    Issue(4325): Define what the correct timeline state is based on scroll
-    offset.
+:   If <var>current scroll offset</var> is less than [=effective start offset=]:
+::  The [=current time=] is 0.
 
-1.  If <var>current scroll offset</var> is greater than or equal to [=effective
-    end offset=], return [=effective time range=].
+:   If <var>current scroll offset</var> is greater than or equal to [=effective
+    end offset=]:
+::  The [=current time=] is the [=effective time range=].
 
-1.  Return the result of evaluating the following expression:
+:   Otherwise, the [=current time=] is the result of evaluating the following
+    expression:
```

Comment:
Nit:

```
:   Otherwise
::  The [=current time=] is the result of evaluating...
```

(i.e. split up the line so it is actually a `<dt>`/`<dd>` pair)